### PR TITLE
Update parser to support edit of emergency contact details

### LINKED ISSUE #69: Update parser to support edit of emergency contact details

### PR DESCRIPTION
**Pull request by charlesliyifeng**
## **Summary of this PR**
- Make room number and emergency contact optional for Person
- Make emergency name and emergency phone optional for EmergencyContact to allow individual changes
- Update parsers and commands to support edit of emergency contact details

Fixes #69

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 1]
```diff
@@ -102,9 +102,9 @@ public void toStringMethod() {
                 + "{name=" + ALICE.getName()
                 + ", phone=" + ALICE.getPhone()
                 + ", email=" + ALICE.getEmail()
-                + ", room number=" + ALICE.getRoomNumber()
+                + ", room number=" + ALICE.getRoomNumber().orElse(null)
```
Comment (EdaYeo) [inline src/test/java/seedu/address/model/person/PersonTest.java:105]: Should either "room number" be changed to "roomNumber" or "emergencyContact" be changed to "emergency contact" for standardization?

**Review by EdaYeo — COMMENTED** [review]
LGTM! Just one small comment regarding standardization. It's quite a big PR so maybe someone else wants to check before we merge?

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -142,6 +162,8 @@ public static class EditPersonDescriptor {
         private Email email;
         private RoomNumber roomNumber;
         private Address address;
+        private Name emergencyName;
+        private Phone emergencyPhone;
```
Comment (SeanFoongjt) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:166]: Should these be emergencyContactName and emergencyContactPhone to make it clearer?

**Review by SeanFoongjt — COMMENTED** [review]
LGTM, just a question on naming.

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (charlesliyifeng) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:166]: No, I don't think so.
I used emergencyName and emergencyPhone to be consistent with the parameter name in the feature specification (EMERGENCY_NAME and EMERGENCY_PHONE).

**Review by SeanFoongjt — APPROVED** [review]

**Pull request merged**